vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/ElementSteps.java

```diff
@@ -90,22 +100,59 @@ public void uploadFile(Locator locator, String filePath) throws IOException
         }
         File fileForUpload = ResourceUtils.isFileURL(resource.getURL()) ? resource.getFile()
                 : unpackFile(resource, filePath);
-        if (descriptiveSoftAssert.assertTrue("File " + filePath + " exists", fileForUpload.exists()))
+        if (descriptiveSoftAssert.assertTrue(String.format(FILE_EXISTS_MESSAGE_FORMAT, filePath),
+                fileForUpload.exists()))
         {
             String fullFilePath = fileForUpload.getAbsolutePath();
             if (isRemoteExecution())
             {
                 webDriverProvider.getUnwrapped(RemoteWebDriver.class).setFileDetector(new LocalFileDetector());
             }
             locator.getSearchParameters().setVisibility(Visibility.ALL);
-            WebElement browse = baseValidations.assertIfElementExists(AN_ELEMENT, locator);
+            WebElement browse = baseValidations.assertIfElementExists("An element", locator);
             if (browse != null)
             {
                 browse.sendKeys(fullFilePath);
             }
         }
     }
 
+    /**
+     * This step uploads a file with the given relative path
+     * <p>A <b>relative path</b> starts from some given working directory,
+     * avoiding the need to provide the full absolute path
+     * (i.e. <i>'about.jpeg'</i> is in the root directory
+     * or <i>'/story/uploadfiles/about.png'</i>)</p>
+     * @param locator The locator for the upload element
+     * @param filePath relative path to the file to be uploaded
+     * @see <a href="https://en.wikipedia.org/wiki/Path_(computing)#Absolute_and_relative_paths"> <i>Absolute and
+     * relative paths</i></a>
+     * @throws IOException If an input or output exception occurred
+     */
+    @When("I select element located by `$locator` and upload `$resourceNameOrFilePath`")
+    public void uploadFile(Locator locator, String filePath) throws IOException
+    {
+        Resource resource = resourceLoader.getResource(ResourceLoader.CLASSPATH_URL_PREFIX + filePath);
+        if (!resource.exists())
+        {
+            resource = resourceLoader.getResource(ResourceUtils.FILE_URL_PREFIX + filePath);
+        }
+        File fileForUpload = ResourceUtils.isFileURL(resource.getURL()) ? resource.getFile()
+                : unpackFile(resource, filePath);
+        if (descriptiveSoftAssert.assertTrue(String.format(FILE_EXISTS_MESSAGE_FORMAT, filePath),
```

Comment:
> each resource is a file

nope

> step naming doesn't look good anymore

we are adhering common naming: https://docs.vividus.dev/vividus/0.5.13/commons/expressions.html#_calculatefilehash

Reply:
1) as for the message: it makes sense to update it and make more descriptive
2) no need to use `softAssert` to check file/resource presence: we can throw exception in case of invalid input - it's considered as user's mistake